prow/plugins/config.go

```diff
@@ -1786,9 +1786,50 @@ func (c *Configuration) mergeFrom(other *Configuration) error {
 		errs = append(errs, fmt.Errorf("failed to merge .bugzilla from supplemental config: %w", err))
 	}
 
+	if err := c.mergeApproveFrom(other.Approve, org, repo); err != nil {
+		errs = append(errs, fmt.Errorf("failed to merge .approve from supplemental config: %w", err))
+	}
+
 	return utilerrors.NewAggregate(errs)
 }
 
+func (c *Configuration) mergeApproveFrom(other []Approve, org, repo string) error {
+	if len(other) == 0 {
+		return nil
+	}
+
+	// If the supplemental configuration points to a specific org/repo, we need to validate that there is
+	// only one approve configuration that holds that org/repo value in the `repos` field.
+	if org != "" && repo != "" {
+		if len(other) > 1 {
+			return fmt.Errorf("supplemental config for %s/%s must have only 1 approve plugin configuration", org, repo)
+		}
+		if !reflect.DeepEqual(other[0].Repos, []string{fmt.Sprintf("%s/%s", org, repo)}) {
+			return fmt.Errorf("the repos field must hold the %s/%s value only", org, repo)
+		}
+	}
+
+	// If the supplemental configuration points to an organization, we need to validate that each of the configurations holds
+	// values in the repo field that belong to that organization.
+	if org != "" && repo == "" {
+		if len(other) > 1 {
```

Comment:
You must not validate anything like that here. The only thing you must validate here is that `other` can be merged. I suppose that is always the case as long as there isn't already a config for the some org or repo.

The checking what orgs and repos are allowed in what file happens in checkonfig through HasConfigFor() which is distinct from the merge code.